### DISCUSSION TITLE
WebSerial REPL Console

### DISCUSSION THREAD

**Discussion by mishal** [micropython/micropython#14186] [Show and tell]
A web console that leverages the WebSerial API to interact with your MicroPython boards directly from your web browser. 
This project eliminates the need for additional software installations. Simply open the web page, connect the device and send commands via the REPL.

The WebSerial API is not something new - there are several similar projects around the web, but I needed something more polished as a part of my larger project. I plan to integrate more features, like graphing, log formatting, device configuration... 

The WebSerial API is available on Chromium based browsers since version 89 - see https://developer.mozilla.org/en-US/docs/Web/API/Navigator/serial#browser_compatibility

## Features:

 * Interact with your MicroPython board directly through your web browser.
 * Send and receive Python commands in real-time.
 * No additional software installation required.
 
## Get started:

Application: https://console.growcube.2000cubits.org
Project Home: https://gitlab.com/growcube/web-console

For more info, see the project homepage. In the meantime, I'd love to hear your thoughts and answer any questions you might have.

**Comment by vshymanskyy** [comment]
Check out `MicroPython Web IDE`: https://github.com/orgs/micropython/discussions/15219
